Build issues for sandybridge builds

@VinInn , last night I tried to build `CMSSW_11_3_SKYLAKEAVX512_X` IB with multiple instruction set ( `haswell, sandybridge and skylake-avx512`). All worked fine for `haswell` and `skylake-avx512` but for  `sandybridge` we get this build error in `TrackingTools/AnalyticalJacobians` [a]. Any idea how to avoid such issue, note that it fails when scram use `-march=sandybridge`


[a] 
https://github.com/cms-sw/cmssw/blob/master/TrackingTools/AnalyticalJacobians/src/AnalyticalCurvilinearJacobian.cc#L48
https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc7_amd64_gcc900/CMSSW_11_3_SKYLAKEAVX512_X_2021-03-24-1100/TrackingTools/AnalyticalJacobians
```
>> Compiling sandybridge TrackingTools/AnalyticalJacobians/src/JacobianLocalToCurvilinear.cc
In file included from gcc/9.3.0/lib/gcc/x86_64-unknown-linux-gnu/9.3.0/include/immintrin.h:53,
                 from gcc/9.3.0/lib/gcc/x86_64-unknown-linux-gnu/9.3.0/include/x86intrin.h:32,
                 from src/DataFormats/Math/interface/SSEVec.h:7,
                 from src/DataFormats/Math/interface/SIMDVec.h:57,
                 from src/DataFormats/GeometryVector/interface/Basic3DVector.h:4,
                 from src/DataFormats/GeometryVector/interface/PV3DBase.h:4,
                 from src/DataFormats/GeometryVector/interface/Point3DBase.h:5,
                 from src/DataFormats/GeometryVector/interface/GlobalPoint.h:5,
                 from src/TrackingTools/AnalyticalJacobians/interface/AnalyticalCurvilinearJacobian.h:5,
                 from src/TrackingTools/AnalyticalJacobians/src/AnalyticalCurvilinearJacobian.cc:1:
gcc/9.3.0/lib/gcc/x86_64-unknown-linux-gnu/9.3.0/include/avx2intrin.h: In function '__m256d mathSSE::_mm256_cross_pd(__m256d, __m256d)':
  gcc/9.3.0/lib/gcc/x86_64-unknown-linux-gnu/9.3.0/include/avx2intrin.h:913:1: error: inlining failed in call to always_inline '__m256i _mm256_xor_si256(__m256i, __m256i)': target specific option mismatch
   913 | _mm256_xor_si256 (__m256i __A, __m256i __B)
      | ^~~~~~~~~~~~~~~~
In file included from src/DataFormats/Math/interface/SIMDVec.h:57,
                 from src/DataFormats/GeometryVector/interface/Basic3DVector.h:4,
                 from src/DataFormats/GeometryVector/interface/PV3DBase.h:4,
                 from src/DataFormats/GeometryVector/interface/Point3DBase.h:5,
                 from src/DataFormats/GeometryVector/interface/GlobalPoint.h:5,
                 from src/TrackingTools/AnalyticalJacobians/interface/AnalyticalCurvilinearJacobian.h:5,
                 from src/TrackingTools/AnalyticalJacobians/src/AnalyticalCurvilinearJacobian.cc:1:
src/DataFormats/Math/interface/SSEVec.h:92:36: note: called from here
   92 |     return __m256d(_mm256_xor_si256(__m256i(ret), neg));

```

--- a/DataFormats/Math/interface/SSEVec.h
+++ b/DataFormats/Math/interface/SSEVec.h
@@ -6,9 +6,9 @@
 #if defined(__GNUC__)
 #include <x86intrin.h>
 #define CMS_USE_SSE
-#ifdef __AVX__
-#define CMS_USE_AVX
-#endif /* __AVX__ */
+#ifdef __AVX2__
+#define CMS_USE_AVX2
+#endif /* __AVX2__ */
 #endif /* defined(__GNUC__) */
 #endif /* !defined(__arm__) && !defined(__aarch64__) && !defined(__MIC__) */
 
@@ -63,7 +63,7 @@ namespace mathSSE {
   }
 #endif  // CMS_USE_SSE
 
-#ifdef CMS_USE_AVX
+#ifdef CMS_USE_AVX2
   inline __m256d __attribute__((always_inline)) __attribute__((pure)) _mm256_dot_pd(__m256d v1, __m256d v2) {
     __m256d mul = _mm256_mul_pd(v1, v2);
     mul = _mm256_hadd_pd(mul, mul);
@@ -92,7 +92,7 @@ namespace mathSSE {
     return __m256d(_mm256_xor_si256(__m256i(ret), neg));
   }
 
-#endif  //  CMS_USE_AVX
+#endif  //  CMS_USE_AVX2
 
   template <typename T>
   struct OldVec {
@@ -101,7 +101,7 @@ namespace mathSSE {
     T theZ;
     T theW;
   } __attribute__((aligned(16)));
-#ifdef CMS_USE_AVX
+#ifdef CMS_USE_AVX2
   template <>
   struct OldVec<double> {
     double theX;
@@ -235,7 +235,7 @@ namespace mathSSE {
     }
   };
 
-#ifdef CMS_USE_AVX
+#ifdef CMS_USE_AVX2
   template <>
   union Mask4<double> {
     __m256d vec;
@@ -380,7 +380,7 @@ namespace mathSSE {
     double operator[](unsigned int n) const { return arr[n]; }
   };
 
-#ifdef CMS_USE_AVX
+#ifdef CMS_USE_AVX2
 }  // namespace mathSSE
 #include "private/AVXVec.h"
 
@@ -473,7 +473,7 @@ namespace mathSSE {
     __m128 v2 = _mm_cvtpd_ps(ivec.vec[1]);
     vec = _mm_shuffle_ps(vec, v2, _MM_SHUFFLE(1, 0, 1, 0));
   }
-#endif  // CMS_USE_AVX
+#endif  // CMS_USE_AVX2
 
 #endif  // CMS_USE_SSE
 
@@ -602,7 +602,7 @@ inline float cross(mathSSE::Vec2F a, mathSSE::Vec2F b) { return a.arr[0] * b.arr
 //
 
 inline mathSSE::Vec2D::Vec2(Vec4D v4) {
-#ifdef CMS_USE_AVX
+#ifdef CMS_USE_AVX2
   vec = _mm256_castpd256_pd128(v4.vec);
 #else
   vec = v4.vec[0];
@@ -671,7 +671,7 @@ inline double cross(mathSSE::Vec2D a, mathSSE::Vec2D b) {
   return s;
 }
 
-#ifndef CMS_USE_AVX
+#ifndef CMS_USE_AVX2
 // double op 3d
 
 #ifdef __SSE3__
@@ -781,7 +781,7 @@ inline double __attribute__((always_inline)) __attribute__((pure)) dotxy(mathSSE
   return dot(a.xy(), b.xy());
 }
 
-#endif  // CMS_USE_AVX
+#endif  // CMS_USE_AVX2
 
 // sqrt
 namespace mathSSE {
@@ -797,7 +797,7 @@ namespace mathSSE {
   inline Vec2D sqrt(Vec2D v) {
     return _mm_sqrt_pd(v.vec);
   }
-#ifdef CMS_USE_AVX
+#ifdef CMS_USE_AVX2
   template <>
   inline Vec4D sqrt(Vec4D v) {
     return _mm256_sqrt_pd(v.vec);

--- a/DataFormats/Math/test/ExtVec_t.cpp
+++ b/DataFormats/Math/test/ExtVec_t.cpp
@@ -5,9 +5,9 @@
 
 #include <iostream>
 
-#ifdef __AVX__
-#define CMS_USE_AVX
-#endif /* __AVX__ */
+#ifdef __AVX2__
+#define CMS_USE_AVX2
+#endif /* __AVX2__ */
 
 void addScaleddiff(Vec3F& res, float s, Vec3F const& a, Vec3F const& b) { res = res + s * (a - b); }
 
@@ -232,12 +232,12 @@ void go(bool dovec = true) {
 }
 
 int main() {
-#ifdef CMS_USE_AVX
+#ifdef CMS_USE_AVX2
   std::cout << "using AVX" << std::endl;
 #endif
   testBa();
   go<float>();
-#ifdef CMS_USE_AVX
+#ifdef CMS_USE_AVX2
   go<double>(false);
 #else
   go<double>();

--- a/DataFormats/Math/test/SSEVec_t.cpp
+++ b/DataFormats/Math/test/SSEVec_t.cpp
@@ -226,12 +226,12 @@ void go(bool dovect = true) {
 }
 
 int main() {
-#ifdef CMS_USE_AVX
-  std::cout << "using AVX" << std::endl;
+#ifdef CMS_USE_AVX2
+  std::cout << "using AVX2" << std::endl;
 #endif
   testBa();
   go<float>();
-#ifdef CMS_USE_AVX
+#ifdef CMS_USE_AVX2
   go<double>(false);
 #else
   go<double>();

--- a/TrackingTools/AnalyticalJacobians/src/AnalyticalCurvilinearJacobianSSE.icc
+++ b/TrackingTools/AnalyticalJacobians/src/AnalyticalCurvilinearJacobianSSE.icc
@@ -50,7 +50,7 @@ void AnalyticalCurvilinearJacobian::computeFullJacobian(const GlobalTrajectoryPa
   Vec4D tt = t * t;
   tt = mathSSE::sqrt(hadd(tt, tt));
 
-#ifdef CMS_USE_AVX
+#ifdef CMS_USE_AVX2
   const __m256d neg = _mm256_setr_pd(-0.0, 0.0, -0.0, 0.0);
   Vec4D res(_mm256_xor_pd(neg, _mm256_div_pd(_mm256_shuffle_pd(t.vec, t.vec, 5), tt.vec)));
   Vec2D u1(res.xy());

